pinot-core/src/main/java/com/linkedin/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -37,7 +38,8 @@
   private final String _clusterName;
 
   // Co-ordinates of the last known controller leader.
-  private volatile String _controllerLeaderHostPort = null;
+  private volatile String _controllerLeaderHostPortNotUsed = null;
+  private volatile Pair<String, Integer> _controllerLeaderHostPort = null;
```

Comment:
No need to be volatile since it's always fetched synchronized

Reply:
good point.